zipline/pipeline/factors/factor.py

```diff
@@ -1227,6 +1227,54 @@ def compute(self, today, assets, out, data):
     '''
     dtype = float64_dtype
 
+    def __getattr__(self, name):
+        if name in self.outputs:
+            return RecarrayFactor(factor=self, attribute=name)
+        else:
+            raise AttributeError(
+                "This factor has no output called '{}'.".format(name)
+            )
+
+    def __iter__(self):
+        if not self.outputs:
+            raise ValueError('This factor does not have multiple outputs.')
+        RecarrayFactor_ = partial(RecarrayFactor, self)
+        return iter(map(RecarrayFactor_, self.outputs))
+
+
+class RecarrayFactor(Factor):
+
+    siblings = {}
+
+    def __new__(cls, factor, attribute):
+        new_instance = super(RecarrayFactor, cls).__new__(
+            cls,
+            factor=factor,
+            attribute=attribute,
+            inputs=factor.inputs,
+            outputs=factor.outputs,
+            window_length=factor.window_length,
+            dtype=factor.dtype,
+            missing_value=factor.missing_value,
+        )
+        cls.siblings.setdefault(factor, set()).add(new_instance)
+        return new_instance
+
+    def _init(self, factor, attribute, *args, **kwargs):
+        self.factor = factor
+        self.attribute = attribute
+        self.compute = factor.compute
+        self._compute = factor._compute
```

Comment:
I'm thinking maybe this should implement its own `_compute` method instead of adopting the parent CustomFactor